lib/aquilon/worker/templates/machine.py

```diff
@@ -40,7 +40,7 @@ class PlenaryMachineInfo(StructurePlenary):
     @classmethod
     def template_name(cls, dbmachine):
         loc = dbmachine.location
-        return "%s/%s/%s/%s/%s" % (cls.prefix, loc.hub.fullname.lower(),
+        return "%s/%s/%s/%s/%s" % (cls.prefix, loc.continent.name,
```

Comment:
We have discussed this with Gabor just now: he suggest to take this even further and get rid of location attributes from the template path: and leave only (cls.prefix, dbmachine.label). Would this work for you?

Reply:
I think the constraint was mainly yours as you have several cities... In general we have only one so we don't need the other level.

Reply:
What about creating a structure using the first letters of the dbmachine.label ?
For instance:

```
return "{0}/{1:.1}/{1:.2}/{1}".format(cls.prefix, dbmachine.label)
```

Reply:
Interesting suggestion, easy to implement... Not sure how diverse the machine names are, in particular their first letter. Ours tend to be something like `rackXXYY` where `XX` is the rack number and `YY` the slot number in the rack. Will defeat the directory hierarchy created from the first letters in our case... but probably not a major issue for us if everybody agrees it is the way to go.

Reply:
Our machine names come from another system and start with one of: `system`, `scarf`, `vm` or `test`; so would have a  similar problem, OTOH we only have 23000 hosts managed by the brokers, so not a huge problem.

Reply:
If we really need to break it up, how about using a modulus of the `hardware_entity_id`?

```python
return "%s/%02x/%s" % (cls.prefix, dbmachine.hardware_entity_id % 256, dbmachine.label)
```

Reply:
@jrha I think it would be good to have it easy to deduce from the machine information. `dbmachine.id / 1000` is probably not the most convenient, even if it is predictible.